files/enc.rb

```diff
@@ -93,44 +117,56 @@ class FactUploadError < StandardError; end
   end
 end
 
-def empty_values_hash?(facts_file)
-  facts = File.read(facts_file)
-  puppet_facts = YAML::load(facts.gsub(/\!ruby\/object.*$/,''))
+def parse_file(filename, mac_address_workaround = false)
+  case File.extname(filename)
+  when '.yaml'
+    data = File.read(filename)
+    quote_macs!(data) if mac_address_workaround && YAML.load('22:22:22:22:22:22').is_a?(Integer)
+    YAML.load(data.gsub(/\!ruby\/object.*$/,''))
+  when '.json'
+    JSON.parse(File.read(filename))
+  else
+    raise "Unknown extension for file '#{filename}'"
+  end
+end
 
+def empty_values_hash?(facts_file)
+  puppet_facts = parse_file(facts_file)
   puppet_facts['values'].empty?
 end
 
 def process_host_facts(certname)
-    f = "#{puppetdir}/yaml/facts/#{certname}.yaml"
-    if File.size(f) != 0
-      if empty_values_hash?(f)
-        puts "Empty values hash in fact file #{f}, not uploading"
-        return 0
-      end
+  f = fact_file(certname)
+  if File.size(f) != 0
+    if empty_values_hash?(f)
+      puts "Empty values hash in fact file #{f}, not uploading"
+      return 0
+    end
 
-      req = generate_fact_request(certname, f)
-      begin
-        upload_facts(certname, req) if req
-        return 0
-      rescue => e
-        $stderr.puts "During fact upload occured an exception: #{e}"
-        return 1
-      end
-    else
-      $stderr.puts "Fact file #{f} does not contain any fact"
-      return 2
+    req = generate_fact_request(certname, f)
+    begin
+      upload_facts(certname, req) if req
+      return 0
+    rescue => e
```

Comment:
I see this code is from before, just wanted to see if my assumption that this implies there is some variety of unknown exception types that might can occur here?

Reply:
Yes, I think the idea is that it shouldn't exit with a stack trace but exit with code 1. There doesn't appear to be an explicit exit though so that probably doesn't work.